test/files/neg/patmat-sealed-reachable.scala

```diff
@@ -0,0 +1,49 @@
+// scalac: -Xfatal-warnings
+
+sealed trait SealedTrait
+
+class SomeClass
+
+trait UnsealedTrait
+
+sealed abstract class O
+
+class O1 extends O
+
+final class Foo
+
+class Bar
+
+class Test {
+
+  // final class, sealed trait
+  def a(c: Option[Int]) = c match { case _: SealedTrait => ; case _ => }
+
+  // final class, SomeClass
+  def b(c: Option[Int]) = c match { case _: SomeClass => ; case _  => }
+
+  // final class, trait, Can't we check it?
+  def c(c: Option[Int]) = c match { case _: UnsealedTrait => ; case _ => }
```

Comment:
This warned in retronym/scala#109, so what changed in 2.13?

Reply:
This is also my problem, so I was sleepy and wrote a note. I hope someone can answer this:  it is not clear where the difference between 2.12 and 2.13 comes from. I don't know what I missed. 😅